dbt/include/bigquery/macros/utils/get_columns_spec_ddl.sql

```diff
@@ -0,0 +1,18 @@
+{% macro bigquery__get_columns_spec_ddl() %}
+  {# loop through user_provided_columns to create DDL with data types and constraints #}
+  {% if config.get('constraints_enabled', False) %}
+    {%- set user_provided_columns = model['columns'] -%}
+    (
+    {% for i in user_provided_columns -%}
+      {%- set col = user_provided_columns[i] -%}
+      {% set constraints = col['constraint'] -%}
```

Comment:
Constraints is a `list`, you can follow the example here to loop through the list of constraints: https://github.com/dbt-labs/dbt-snowflake/pull/341/files#r1039792937